core-bundle/src/Resources/contao/library/Contao/Message.php

```diff
@@ -178,7 +178,15 @@ public static function reset()
 			return;
 		}
 
-		$session->getFlashBag()->clear();
+		$flashBag = $session->getFlashBag();
+
+		// Find all contao.BE. keys (see #3393)
+		$keys = preg_grep('(^contao\.BE\.)', $flashBag->keys());
```

Comment:
`FE` is possible too, right? Probably not used by the core but if you use `Message::add()` from within the FE, it would now not get cleared anymore which is the most important part due to the session/caching.

Reply:
Agreed. Should we maybe remove all keys starting with `contao.`?

Reply:
I think that makes sense, yes. We use the prefix already in the `getFlashBagKey()` method. Make that prefix a class constant and use it in both methods, then that should be fine :)

Reply:
May be only depending on current scope? An action that clears the flash bag for the back end should not also clear the flash bag for the front end, in case you are logged into the back end and also visit the front end with the same session.

Reply:
The scope can be anything, because all `add*()` and `generate*()` methods allow to set a custom scope. Therefore removing all keys starting with `contao.` is the only way to reliably reset the messages.